### PR TITLE
#396 Ouput slack node in separate file.

### PR DESCRIPTION
**Pull request by gautierbureau**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [ ] unit tests and non regression tests were added (update of inputs, outputs, SA, algos)
- [ ] main documentation was updated (update of input/output file)
- [ ] if this commit targets a release branch: the corresponding milestone was added in the ticket and in this PR

### CODE REVIEW

**Comment by gautierbureau** [comment]
I updated with a json file

**Comment by gautierbureau** [comment]
Linked powsybl PR : https://github.com/powsybl/powsybl-dynawo/pull/106

File: sources/Context.cpp [hunk 1]
```diff
@@ -344,6 +345,37 @@ Context::executeSecurityAnalysis() {
   saLauncher->writeResults();
 }
 
+void
+Context::exportResults(bool isOk) {
+  switch (def_.simulationKind) {
+  case SimulationKind::STEADY_STATE_CALCULATION: {
+    boost::property_tree::ptree resultsTree, componentResultsTree, componentResultsChild;
+    resultsTree.put("version", "1.2");
```
Comment (rosiereflo) [inline sources/Context.cpp:353]: Is this the powsybl version? If so we are using 1.4

File: sources/Context.h [hunk 2]
```diff
@@ -106,6 +106,15 @@ class Context {
    */
   void exportOutputs();
 
+  /**
+   * @brief Export results file
+   *
+   * @param isOk the simulation was ok
```
Comment (rosiereflo) [inline sources/Context.h:112]: Maybe we could call this parameter "simulationOk"?

**Comment by rosiereflo** [comment]
There is also 1 code smell to fix

File: sources/Context.cpp (hunk 1 above)
Comment (gautierbureau) [reply sources/Context.cpp:353]: No this is the version of the LodFlowResult format. See https://github.com/powsybl/powsybl-core/blob/main/loadflow/loadflow-api/src/test/resources/LoadFlowResultVersion12.json

File: sources/Context.cpp [hunk 3]
```diff
@@ -344,6 +345,37 @@ Context::executeSecurityAnalysis() {
...
+    resultsTree.put("isOK", isOk);
```
[the remark this answers is not in the document]
Comment (gautierbureau) [reply sources/Context.cpp:354]: it is mandatory in the result format

File: sources/Context.h (hunk 2 above)
Comment (gautierbureau) [reply sources/Context.h:112]: we can but the name of the parameter in the json needs to be isOk

**Comment by gautierbureau** [comment]
> There is also 1 code smell to fix

If fixed it and solved other issues

File: CMakeLists.txt [hunk 4]
```diff
@@ -64,7 +64,7 @@ set(FORCE_CXX11_ABI OFF CACHE BOOL "Choose either ON or OFF.")
 
 option(DYNAFLOW_LAUNCHER_USE_ADDITIONAL_SOURCEFILE "Use the additional source file" ON)
 
-option(DYNAFLOW_LAUNCHER_SHARED_LIB "Compile shared library" OFF)
+option(DYNAFLOW_LAUNCHER_SHARED_LIB "Compile shared library" ON)
```
Comment (rosiereflo) [inline CMakeLists.txt:67]: Not related, should be done in another PR

**Review by rosiereflo — CHANGES_REQUESTED** [review]

**Review by rosiereflo — APPROVED** [review]
